### PR TITLE
update pybullet_setJointMotorControl with seperate args for targetPosition, targetVelocity, kp and kd

### PR DESCRIPTION
**Pull request by matpalm**
( based on yesterday's b3RobotSimAPI.cpp change )

### CODE REVIEW

File: examples/pybullet/pybullet.c [hunk 1]
```diff
@@ -357,27 +371,26 @@ static PyObject* pybullet_setJointMotorControl(PyObject* self, PyObject* args)
...
+
         switch (controlMode)
         {
             case CONTROL_MODE_VELOCITY:
             {
-				double kd = gains;
                 b3JointControlSetDesiredVelocity(commandHandle, info.m_uIndex, targetValue);
-                b3JointControlSetKd(commandHandle,info.m_uIndex,kd);
-                b3JointControlSetMaximumForce(commandHandle,info.m_uIndex,maxForce);
+                b3JointControlSetKd(commandHandle, info.m_uIndex, kd);
```
Comment (matpalm) [inline examples/pybullet/pybullet.c]: oh wait, with CONTROL_MODE_VELOCITY kd isn't configurable. more strongly thinking that this being three methods would be much cleaner; there's a mix of args that are/aren't use by the different control modes and most of this code is checking the combos...

**Comment by erwincoumans** [comment]
Thanks for the patch, I'll look at it.
 //TODO(matkelcey): should this just be three methods?
One idea is to use optional named arguments rather than additional methods.

**Comment by matpalm** [comment]
i think the trouble is it's more than named args with sensible defaults; some args just aren't applicable to some control modes.... e.g. CONTROL_MODE_TORQUE only uses the target & doesn't care about max force, kd or kp.

**Comment by erwincoumans** [comment]
This is mainly a documentation issue: pybullet follows the underlying C-API. Arguments that are not applicable for a certain mode are ignored. It is possible to protect the user, report errors/warnings and add convenience functions in object-oriented wrappers around pybullet,

**Comment by matpalm** [comment]
ah yes, true. following the c-api is an important point that had slipped my mind...

**Comment by erwincoumans** [comment]
Once we introduce named arguments, we can get rid of a lot of clutter that check for the number of args etc. It will break the API though, but at this still it is still very experimental anyway.

**Comment by erwincoumans** [comment]
The default kd when using CONTROL_MODE_VELOCITY should be 1.0, not 0.1, when not specified. 

It is not due to this pull request, but there is an issue with defaults values not overridden by the user. When the user does NOT explicitly override the kd/kp values, pybullet should not come up with some default, but it should NOT call the function to override it. That way, the defaults are maintained in one place, in the C-API. So we need to fix that soon.

**Comment by erwincoumans** [comment]
Thanks for the help, and hopefully we sort out this named arguments stuff soon!

**Pull request merged**